nuxeo-core/nuxeo-core-mongodb/src/main/java/org/nuxeo/ecm/core/mongodb/seqgen/MongoDBUIDSequencer.java

```diff
@@ -101,6 +104,20 @@ public void initSequence(String key, long id) {
 
     @Override
     public long getNextLong(String key) {
+        return incrementBy(key, ONE);
+    }
+
+    @Override
+    public List<Long> getNextBlock(String key, int blockSize) {
+        List<Long> ret = new ArrayList<>(blockSize);
+        long last = incrementBy(key, Long.valueOf(blockSize));
+        for (int i = blockSize - 1; i >= 0; i--) {
+            ret.add(last - i);
+        }
+        return ret;
+    }
+
+    protected long incrementBy(String key, Long value) {
```

Comment:
It's strange to have `Long` here. You should use `int` and just convert to `Long` when actually needed in `sequence.put`.